gfx/polygon.go

```diff
@@ -264,7 +264,7 @@ func (p *polygon) updateBounds() {
 			polygonY2 = ybottom
 		}
 		edgeX1 := edge.xstart
-		edgeX2 := edge.xstart + (ybottom-ytop)*edge.xinc
+		edgeX2 := edge.xstart + (ybottom-ytop)*(edge.xinc/4)
```

Comment:
Yeah I think I see the bug.

If you look at the definition of `xinc` you can see that it has the comment `// .16`. I don't know whether I'm explaining that anywhere, but it means `xinc` is a fixed point value with 16 fractional bits. If you follow the value, it eventually ends up in `boundX1` or `boundX2` shifted right by 16, stripping off those 16 fractional bits. So `xinc` is correct.

The real bug is something different. `ybottom` and `ytop` both have two fractional bits (as indicated with the `// .2` comment). `ybottom-ytop` is a subtraction, so they keep the number of fractional bits. But then they get multiplied by `xinc` (with 16 fractional bits) meaning `edgeX2` ends up adding `xstart` (that has 16 fractional bits) to a value with 18 fractional bits. That's incorrect!

A better fix would be the following:

```suggestion
		edgeX2 := edge.xstart + ((ybottom-ytop)*edge.xinc) / 4
```


This makes sure the value with 18 fractional bits gets scaled back to 16 bits before adding to `xstart`. Doing it after the multiplication means the loss in precision is reduced (at the risk of integer overflows, which would only happen with 2**14 = 16K displays so I think it's fine).

Reply:
This is indeed similar to a line like this:

https://github.com/aykevl/tinygl/blob/ef85831fead9a68811df19bf213427bb0d03edd0/gfx/polygon.go#L50

`xinc` (which has 16 fractional bits) is multiplied with `edgeY` (which has 2 fractional bits) and then the result is divided by 4 to get back to 16 fractional bits.

(Note that I used the same comment convention here: `// .0` means the result has no fractional bits - it's just a pixel value).

Reply:
Alternatively you could leave out the parentheses to be consistent with the other code (this is equivalent to the earlier suggestion I posted):

```suggestion
		edgeX2 := edge.xstart + (ybottom-ytop)*edge.xinc/4
```

Reply:
Thanks for the technical details. 
I'll make some tests with your suggestion.

Reply:
I updated the PR with your last suggestion. 
Works properly with my gauge demo application.